ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -1138,9 +1164,69 @@ doAllocBuffers(unsigned bufferSize,
     rstKeywords["PRES"] = 0;
     rstKeywords["PRESSURE"] = 0;
 
-    // If TEMP is set in RPTRST we output temperature even if THERMAL
-    // is not activated
-    if (enableEnergy_ || rstKeywords["TEMP"] > 0) {
+    if (enableMech_ && eclState_.runspec().mech()){
+        this->mechPotentialForce_.resize(bufferSize,0.0);
+        rstKeywords["MECHPOTF"] = 0;
```

Comment:
Unless we expect users to add these quantities to the `RPTRST` request, we don't really need to add the `rstKeyword["something"] = 0` statements here.